azavea/pfb-network-connectivity
# Remove layer filter

## Overview

Removes the querystring-based layer filter functionality, since it's not being used and isn't compatible with the intended caching strategy.

### Notes

- The immediate impetus for removing it (rather than letting it sit there, which it was doing before without hurting anything) is that it turns out `claudia-local-api` and the actual API Gateway proxy's request objects are different in the case of a request with no querystring--`claudia-local-api` provides an empty object, but API Gateway sets `req.queryStringParameters` to null.  This caused a crash on Lambda that wasn't happening locally.  And yeah, any deviation from "just sitting there not getting in the way" is cause enough to rip this out.
- This takes the implementation of Tilegarden here farther from the upstream Tilegarden project, but this implementation was always meant to be customized, and it's already far enough from upstream that there's no question of merging or large-scale copying in either direction.  Which is fine.

## Testing Instructions

This removes some tests, since there was a whole test file for the layer filtering.  Since the issue that prompted this only showed up on AWS, local testing only goes so far.  It makes sense to push this to staging manually and confirm that the analysis tiles show up.  Which can be accomplished by pulling down the `.env` and `claudia.json` from the Staging settings bucket [like in `scripts/infra`](https://github.com/azavea/pfb-network-connectivity/blob/develop/scripts/infra#L83-L84) and running the Tilegarden deployment command (`docker-compose -f docker-compose.yml -f docker-compose.test.yml run --rm --entrypoint yarn tilegarden deploy`).

Connects #634 and #635
